### ISSUE TITLE
Naga_oil breaks when importing a struct with a field name ending in an underscore

### ISSUE THREAD

**Issue by Aaron1011** [robtfm/naga_oil#15]
I've modified one of the example shaders to demonstrate the issue: https://github.com/Aaron1011/naga_oil/commit/d9e8cd204b52eacb29fbe48168fcc6b4d5fcd226
Running `cargo run --example pbr_compose_test` produces the following output:

```
running 1000 full composer builds (no caching)
error: invalid field accessor `myfield_`
    ┌─ examples/bevy_pbr_wgsl/pbr_lighting.wgsl:286:24
    │
286 │     let myfield_ = val.myfield_;
    │                        ^^^^^^^^ invalid accessor
    │
    = invalid field accessor `myfield_`
```

Renaming both the field definition and usage to `myfield_a` (or anything that doesn't end in an underscore) allows the example to run successfully.

**Comment by robtfm** [comment]
this is a bit awkward. 

when writing modules out into native shader languages, naga [converts identifiers](https://github.com/gfx-rs/naga/blob/1be8024bda3594987b417bead5024b98be9ab521/src/proc/namer.rs#L32) so that the resulting names are conformant with all target languages. this includes removing trailing underscores, but also covers things like remapping reserved glsl keywords when you're writing a wgsl shader, etc.

in naga_oil, for language interoperability i take the import source, compile to naga, strip out function bodies then write out a header string for the target language. then when building a final shader, i string-join the header with the final shader to get something i can compile to naga, which is then combined with the real (non-stripped) IR of the import to get a final naga module.

because the identifier has been renamed in the header, the string-joined shader doesn't build to naga... and robustly detecting identifiers that have been renamed and substituting them in the string-joined shader is difficult without being able build a naga module in the first place (i would need AST context at least).

i think currently the best i can do is to fail a bit more gracefully, by rebuilding the header string to naga and checking that all the original identifiers are preserved, and throwing an error if not.

**Comment by Aaron1011** [comment]
Explicitly rejecting field names ending in an underscore would be fine by me.

**Comment by robtfm** [comment]
fixed in e714ed0

nb - only works with `Composer::default()`, the check isn't enabled in `Composer::non_validating()`

thanks for reporting!
